commit 008003dc6e83439c5e04a744b7fd8197df19096e
Author: H.J. Lu <hjl.tools@gmail.com>
Date:   Sat Jan 29 05:22:31 2022 -0800

    tst-socket-timestamp-compat.c: Check __TIMESIZE [BZ #28837]
    
    time_t size is defined by __TIMESIZE, not __WORDSIZE.  Check __TIMESIZE,
    instead of __WORDSIZE, for time_t size.  This fixes BZ #28837.
    
    (cherry pick from commit 77a602ebb0769e7ccc5f9f8e06f7fffe66f69dfc)

diff --git a/sysdeps/unix/sysv/linux/tst-socket-timestamp-compat.c b/sysdeps/unix/sysv/linux/tst-socket-timestamp-compat.c
index de261dae5a6385cf..0ff1a214e605105b 100644
--- a/sysdeps/unix/sysv/linux/tst-socket-timestamp-compat.c
+++ b/sysdeps/unix/sysv/linux/tst-socket-timestamp-compat.c
@@ -237,7 +237,7 @@ do_test (void)
 {
   /* This test only make sense for ABIs that support 32 bit time_t socket
      timestampss.  */
-  if (sizeof (time_t) > 4 && __WORDSIZE == 64)
+  if (sizeof (time_t) > 4 && __TIMESIZE == 64)
     return 0;
 
   srv = xsocket (AF_INET, SOCK_DGRAM, 0);
